### PR TITLE
kube,cmd/{k8s-operator,containerboot},envknob,ipn/store/kubestore,*/depaware.txt: split out shared kube types further

### PR DESCRIPTION
**Pull request by irbekrm**
Further split kube package into:

```yaml
- kube
   - types
   - client
   - api
 ```

The ./kube package is intended to contain minimal types that can be shared with non-kube Tailscale libraries/binaries.
However since https://github.com/tailscale/tailscale/pull/13410 the functionality in ./kube got shared by even more non-kube components, who don't need the kube client and API types- so split the package further.

Updates#cleanup

### CODE REVIEW

File: envknob/envknob.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ import (
 	"sync/atomic"
 	"time"
 
-	"tailscale.com/kube"
+	kubetypes "tailscale.com/kube/types"
```
Comment (irbekrm) [inline envknob/envknob.go:34]: Being able to do this (and so not import the kube client/API types) is the main goal of this PR

**Review by maisem — APPROVED** [review]

**Comment by bradfitz** [comment]
Rather than rename the package on import, maybe rename the package overall? Like std has http/httputil and io/ioutil and not io/util and http/util.

**Comment by irbekrm** [comment]
> Rather than rename the package on import, maybe rename the package overall? Like std has http/httputil and io/ioutil and not io/util and http/util.

I wasn't sure which is less bad in terms of stutter, to have `kube/kubetypes` etc and then pull them in `"tailscale.com/kube/kubetypes"` or to have `kube/types` and pull them in as `kubetypes "tailscale.com/kube/types"`.
Happy to rename if it seems to sound better

**Comment by irbekrm** [comment]
@bradfitz https://github.com/tailscale/tailscale/pull/13418

**Pull request merged**